runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/expansion/ExpansionService.java

```diff
@@ -77,7 +78,7 @@
    * @param <InputT> input {@link PValue} type of the transform
    * @param <OutputT> output {@link PValue} type of the transform
    */
-  public interface TransformProvider<InputT extends PValue, OutputT extends PValue> {
+  public interface TransformProvider<InputT extends PInput, OutputT extends PValue> {
```

Comment:
This limit cross language pipelines to sources only. Shall we keep them general?

Reply:
`PValue` also is a `PInput`. So this does not limit in any way, it just allows root transforms to be translated.